src/mscorlib/src/System/Diagnostics/Stacktrace.cs

```diff
@@ -243,11 +243,27 @@ private void OnDeserialized(StreamingContext context)
         }
     }
 
+    [Flags]
+    internal enum StackTraceFormattingOptions
+    {
+        None                                 = 0,
+        ExcludeStackTraceHiddenAttribute     = 1 << 0,
```

Comment:
>ExcludeStackTraceHiddenAttribute [](start = 8, length = 32)

I like this enum. I wonder though, if we really need a new attribute now that we have an enum value specifying what should the behavior be. Having ExcludeNonUserCode flag and using DebuggerNonUserCodeAttribute would imo be better. Someone mentioned this attribute may already be used by some code generators. That's actually an argument for reusing the attribute. The intention of this attribute is to hide helpers that clutter debugging experience. To me that's look like what we are trying to do here as well. If we add a new attribute such generator will need to be updated to emit that attribute as well. Since we have a new flag that determines the stack formatting behavior it's not a breaking change to use the existing attribute.

Reply:
I think that will work very well; also can tie up with the statemachine back pointers (and other generated code)

Reply:
I could accept having an ExcludeNonUserCode flag on this enum, but at this point I have strong concerns about making it the default or considering it a replacement for the StackTraceHidden attribute. Pretty much same logic as above:
 a) The msdn docs make it clear that DebuggerNonUserCode attribute doesn't have any runtime semantics. I don't want to make breaking changes on an unvalidated assumption that this is what users of that attribute would prefer. 
 b) Jan's example was excellent - .Net Native gave us a very strong example that users can be perfectly happy when VS hides something from them with JMC, but they complained extensively when the runtime hid that same thing from them in the exception callstack. 

Its possible with a bunch of research about what code would be affected and talking to customers to see how they feel about those particular classes of methods we could ameliorate those concerns - but it strikes me as a lot of work and a lot of risk for a relatively small potential upside. 

I still think we're better off sticking to the original ExcludeStackTraceHiddenAttribute and not even adding the ExcludeNonUserCodeAttribute until we've got a concrete customer pointing at a specific use case where the ability to customize that would help them.

Reply:
> strong concerns about making it the default 

I didn't mean it to be the default. On contrary. We should keep the default behavior as is. Having an option to treat DebuggerNonUserCode differently in StackTrace is not a breaking change. We can document that this flag exists on the attribute.

If it is not a default and the code that displays the stack trace needs to pass in a new flag then (and only then) I believe adding a new attribute is not useful. It'd just add more complexity.